fcm-channel/src/main/java/io/fcmchannel/sdk/services/FcmClientRegistrationIntentService.java

```diff
@@ -38,22 +38,22 @@ protected void onHandleIntent(Intent intent) {
         try {
             String urn = intent.getStringExtra(EXTRA_URN);
             String contactUuid = intent.getStringExtra(EXTRA_CONTACT_UUID);
-
-            String fcmToken = FirebaseInstanceId.getInstance().getToken();
+            String fcmToken = FcmClient.getFcmToken();
 
             Preferences preferences = FcmClient.getPreferences();
             Contact contact = null;
 
-            if (!TextUtils.isEmpty(FcmClient.getToken())) {
-                Response<FcmRegistrationResponse> response = saveContactWithToken(urn, fcmToken, contactUuid);
+            Response<FcmRegistrationResponse> response = saveContactWithToken(urn, fcmToken, contactUuid);
+
+            if (response.isSuccessful()) {
+                contact = new Contact();
```

Comment:
where this contact object is used?

Reply:
Here: https://github.com/push-platform/fcm-channel-android/pull/35/files#diff-a2526e0d221caae34088316e4a04b744R81

That way the client lib can extends this service and override onFcmRegistered to perform some operation, like: https://github.com/push-platform/fcm-channel-android/blob/master/sample/src/main/java/io/fcmchannel/sdk/sample/services/PushRegistrationService.java#L18